responsibleai/model_monitor/components/src/calculate_attribution.py

```diff
@@ -0,0 +1,177 @@
+
+_logger = logging.getLogger(__file__)
+_ai_logger = None
+
+
+def _get_logger():
+    global _ai_logger
+    if _ai_logger is None:
+        _ai_logger = _LoggerFactory.get_logger(__file__)
+    return _ai_logger
+
+
+_get_logger()
+
+
+def create_lightgbm_model(X, y, task_type):
+    """Create model on which to calculate feature importances
+
+    Args:
+      X: list of x values (data excluding target columns)
+      y: list y values (target column data)
+      task_type: str, the task type (regression or classification) of the resulting model
+
+    Returns:
+       An appropriate model.
+    """
+    if (task_type == "classification"): 
+        lgbm = LGBMClassifier(boosting_type='gbdt', learning_rate=0.1,
+                              max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    else: 
+        lgbm = LGBMRegressor(boosting_type='gbdt', learning_rate=0.1,
+                            max_depth=5, n_estimators=200, n_jobs=1, random_state=777)
+    model = lgbm.fit(X, y)
+    
+    _logger.info("Created lightgbm model")
+    return model
+
+
+def get_model_wrapper(task_type, target_column, baseline_df, production_df):
+    """Create model wrapper using ml-wrappers on which to calculate feature importances
+
+    Args:
+      task_type: str, The task type (regression or classification) of the resulting model
+      target_column: str, the column to predict
+      baseline_df: The baseline dataframe
+      production_df: The production dataframe
+
+    Returns:
+       An appropriate model wrapper
+    """
+    y_train = baseline_df[target_column]
+    x_train = baseline_df.drop([target_column], axis=1)
+    x_test = production_df.drop([target_column], axis=1)
+    model = create_lightgbm_model(x_train, y_train, task_type)
+
+    if task_type == 'classification':
+        all_data = pd.concat([x_test, x_train])
+        model_predict = model.predict(all_data)
+        model_predict_proba = model.predict_proba(all_data)
+        model_wrapper = PredictionsModelWrapperClassification(
+            all_data,
+            model_predict,
+            model_predict_proba)
+    else:
+        all_data = pd.concat([x_test, x_train])
+        model_predict = model.predict(all_data)
+        model_wrapper = PredictionsModelWrapperRegression(all_data, model_predict)
+
+    _logger.info("Created ml wrapper")
+    return model_wrapper
+
+
+def compute_categorical_features(baseline_df, target_column):
+    """Compute which features are categorical based on data type of the columns.
+
+    Args:
+      baseline_df: The baseline dataframe
+      target_column: str, the column to predict
+
+    Returns:
+      list: A list of categorical features.
+    """
+    categorical_features = []
+    for column in baseline_df.columns:
+        baseline_column = pd.Series(baseline_df[column])
+        if baseline_column.dtype.name == 'object' and baseline_column.name != target_column:
```

Comment:
what about the "category" dtype?

Reply:
what do you mean ?

Reply:
These are pandas types, right? `"object"` is a common one, the way I understand it it's kind of the "kitchen sink"/catch-all type, but `"category"` is actually meant for categorical features. That's why I was wondering why `"category"` isn't mentioned here.

Reply:
discussed offline; seems like using dtype is generally the way to determine the type and we'd have to classify columns as category which is a bit of overkill in this instance because the categories themselves aren't necessary